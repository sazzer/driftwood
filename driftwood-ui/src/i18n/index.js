// @flow

import i18n from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import defaultTranslations from './messages.json';

i18n
    .use(LanguageDetector)
    .init({
        resources: {
            dev: {
                driftwood: defaultTranslations
            }
        },

        // have a common namespace used around the full app
        ns: ['driftwood'],
        defaultNS: 'driftwood',

        debug: false,

        interpolation: {
            escapeValue: false
        },

        parseMissingKeyHandler: (key) => `!!${key}!!`
    });

export default i18n;
