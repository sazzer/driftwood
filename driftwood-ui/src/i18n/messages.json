{
  "page": {
    "title": "Driftwood"
  },
  "authentication": {
    "menu": {
      "title": "Log In",
      "facebook" : "Facebook",
      "google": "Google",
      "twitter": "Twitter"
    }
  }
}
