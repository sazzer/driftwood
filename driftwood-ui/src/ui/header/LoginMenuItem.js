// @flow

import React from 'react';
import {NamespacesConsumer} from 'react-i18next';
import {Dropdown} from 'semantic-ui-react';

/**
 * The props for the Login Menu Item component
 */
type LoginMenuItemProps = {
    provider: string
}

/**
 * The component representing a single item on the login menu
 * @constructor
 */
export function LoginMenuItem({provider}: LoginMenuItemProps) {
    return (
        <NamespacesConsumer>
            {
                (t) => <Dropdown.Item icon={provider} data-provider={provider} text={t(`authentication.menu.${provider}`)} />
            }
        </NamespacesConsumer>

    );
}

export default LoginMenuItem;
