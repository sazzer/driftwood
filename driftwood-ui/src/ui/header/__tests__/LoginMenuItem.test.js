// @flow

import React from 'react';
import {render} from 'enzyme';
import LoginMenuItem from '../LoginMenuItem';
import i18n from "../../../i18n";
import {I18nextProvider} from "react-i18next";

/** Set up the component to test */
function setup(provider: string) {
    return {
        element: render(<I18nextProvider i18n={ i18n }><LoginMenuItem provider={provider} /></I18nextProvider>)
    };
}

it('Renders correctly', () => {
    const { element } = setup('google');

    expect(element).toMatchSnapshot();
});
