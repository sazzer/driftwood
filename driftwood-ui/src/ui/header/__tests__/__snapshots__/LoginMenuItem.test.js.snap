// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  class="item"
  role="option"
>
  <i
    aria-hidden="true"
    class="google icon"
  />
  <span
    class="text"
  >
    Google
  </span>
</div>
`;
